app/src/main/java/org/mozilla/fenix/library/bookmarks/BookmarkFragmentInteractor.kt

```diff
@@ -0,0 +1,172 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.library.bookmarks
+
+import android.content.Context
+import android.view.View
+import androidx.navigation.NavController
+import mozilla.components.concept.storage.BookmarkNode
+import mozilla.components.concept.storage.BookmarkNodeType
+import org.mozilla.fenix.BrowserDirection
+import org.mozilla.fenix.BrowsingModeManager
+import org.mozilla.fenix.HomeActivity
+import org.mozilla.fenix.R
+import org.mozilla.fenix.components.FenixSnackbar
+import org.mozilla.fenix.components.metrics.Event
+import org.mozilla.fenix.components.metrics.MetricController
+import org.mozilla.fenix.ext.asActivity
+import org.mozilla.fenix.ext.components
+import org.mozilla.fenix.ext.copyUrl
+import org.mozilla.fenix.ext.nav
+
+@SuppressWarnings("TooManyFunctions")
+class BookmarkFragmentInteractor(
+    private val context: Context,
+    private val navController: NavController,
+    private val bookmarkStore: BookmarkStore,
+    private val sharedViewModel: BookmarksSharedViewModel,
+    private val deleteBookmarkNodes: (Set<BookmarkNode>, Event) -> Unit
+) : BookmarkInteractor, SignInInteractor {
+
+    val activity: HomeActivity?
+        get() = context.asActivity() as? HomeActivity
+    val metrics: MetricController
+        get() = context.components.analytics.metrics
+
+    override fun change(node: BookmarkNode) {
+        bookmarkStore.dispatch(BookmarkAction.Change(node))
+    }
+
+    override fun open(item: BookmarkNode) {
+        if (item.type == BookmarkNodeType.ITEM) {
+            item.url?.let { url ->
+                activity!!
+                    .openToBrowserAndLoad(
+                        searchTermOrURL = url,
+                        newTab = true,
+                        from = BrowserDirection.FromBookmarks
+                    )
+            }
+        }
+        metrics.track(Event.OpenedBookmark)
```

Comment:
Having the metrics call outside of this made me wonder, is it possible to `open` something that isn't a BookmarkNodeType.ITEM?

Should we make that a requirement for the parameter?

Reply:
We only allow opening items today. As a UX decision, I believe we settled on not allowing opening a folder, either recursively or on one level. I'll switch it to a `require()` call instead.